src/systemdhook.c

```diff
@@ -793,38 +790,19 @@ int main(int argc, char *argv[])
 		snprintf(config_file_name, PATH_MAX, "%s", msg);
 	}
 
-	/* Parse the config file */
-
 	if (fp == NULL) {
 		pr_perror("Failed to open config file: %s", config_file_name);
 		return EXIT_FAILURE;
 	}
 
-	if (fstat(fileno(fp), &configData_stat) == -1) {
-		pr_perror("Failed to determine size of config file: %s", config_file_name);
-		return EXIT_FAILURE;
-	}
-	if (configData_stat.st_size == 0) {
-		pr_perror("config file %s is empty", config_file_name);
-		return EXIT_FAILURE;
-	}
-	configData_size = configData_stat.st_size;
-
-	configData = (char *)malloc((size_t)configData_size+1);
+	/* Read the entire config file */
+	configData = getJSONstring(fp, (size_t)CHUNKSIZE, "config data");
 	if (configData == NULL) {
-		pr_perror("Failed to allocate buffer for config data: size=%ld", configData_size+1);
+		pr_perror("failed to read config data from %s", config_file_name);
```

Comment:
Why not just pass this message into the getJSONstring msg field? Then you would just get one error message in the log not two.

Then just return EXIT_FAILURE.